.github/workflows/docs.yml

```diff
@@ -35,19 +35,47 @@ jobs:
       - uses: dtolnay/rust-toolchain@7b1c307e0dcbda6122208f10795a713336a9b35a
         with:
           toolchain: stable
+      - uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84
       - uses: actions/checkout@v4
       - name: Install mdBook
         run: |
           cargo install --version ${MDBOOK_VERSION} mdbook
       - name: Install mdBook Link-Check
         run: |
           cargo install mdbook-linkcheck
+      - name: Install Scarb
+        uses: software-mansion/setup-scarb@v1
```

Comment:
https://github.com/DelevoXDG/scarb/actions/runs/11443488714/job/31836249298

Tried doing that for corelib docs generation already. For now it fails, reason tbd, I'll check if it's the same with foundry.

Reply:
There was no issue https://github.com/DelevoXDG/starknet-foundry/actions/runs/11443688802/workflow, so we can use nightlies for generating docs for these libs. Although I wonder if it's possible at any point that scarb doc included with nightly versions can be incompatible with previous stable versions of Scarb? @wawel37

Reply:
@DelevoXDG We can change this workflow to rely on stable after 2.9 is out. Right now, it's worth running the nightly. In general, nightly should not introduce breaking changes (until Scarb does with the next release of course), but may include some bugs.

Reply:
Ideally, we want to keep the scarb doc as much backward compatible as we can. Tbh I don't have any example of a future change, that could make such a breaking change. As @maciektr  said, i think you can use the nightly for now. In worst case scenario, it could contain some bugs

Reply:
Please add an issue to replace this with a stable version once it's out.

Reply:
Done: https://github.com/foundry-rs/starknet-foundry/issues/2607